kotlinx-coroutines-core/native/src/EventLoop.kt

```diff
@@ -15,7 +15,8 @@ internal actual abstract class EventLoopImplPlatform : EventLoop() {
     }
 
     protected actual fun reschedule(now: Long, delayedTask: EventLoopImplBase.DelayedTask) {
-        DefaultExecutor.invokeOnTimeout(now, delayedTask, EmptyCoroutineContext)
+        val delayTimeMillis = delayNanosToMillis(nanoTime() - now)
+        DefaultExecutor.invokeOnTimeout(delayTimeMillis, delayedTask, EmptyCoroutineContext)
```

Comment:
I can think of this simple test:
```kotlin
    /** Tests that the delayed tasks scheduled on a closed `runBlocking` event loop get processed in reasonable time. */
    @Test
    fun testReschedulingDelayedTasks() {
        val job = runBlocking {
            val dispatcher = coroutineContext[ContinuationInterceptor]!!
            GlobalScope.launch(dispatcher) {
                delay(1.milliseconds)
            }
        }
        runBlocking {
            withTimeout(10.seconds) {
                job.join()
            }
        }
    }
```
This would miss the error in the opposite direction (like immediately executing the tasks instead), but testing that is trickier, so maybe let's leave it at that?

A good file for this test could be `kotlinx-coroutines-core/concurrent/test/RunBlockingTest.kt`.

Reply:
Added the test. It failed before the fix and succeeds now.